src/generate_east_asian_width.py

```diff
@@ -4,30 +4,55 @@
 import unicode_data_property
 
 
-def output_east_asian_width_data(leading_comment,
-                                 east_asian_width_properties,
-                                 skip=''):
-    print('/* The content of this file is generated from:')
-    print(leading_comment, end='')
-    print('*/')
-    print('')
-    print('#include "eastasianwidthdef.h"')
-    print('')
-    print('static const struct EastAsianWidthProperties eaw_prop[] = {')
+def lookup_east_asian_width(cp, east_asian_width_properties):
     for start, (end, prop) in east_asian_width_properties.items():
-        if prop == skip:
-            continue
-        print(f"    {{0x{start:04X}, 0x{end:04X}, EAW_{prop}}},")
-    print('};')
+        if cp >= start and cp <= end:
+            return prop
+
+
+def generate_function(line_break_properties, east_asian_width_properties):
```

Comment:
`generate_function` is a bit general. I prefer something like `output_is_op_east_asian_function`.